tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -848,6 +812,118 @@ def solve_tridiag(
     np.allclose(expected, out)
 
 
+def test_ternary_operator():
+    size = 10
+
+    a = np_as_located_field(IDim)(2 * np.ones((size,)))
+    b = np_as_located_field(IDim)(2 * np.ones((size,)))
+    out = np_as_located_field(IDim)(np.zeros((size,)))
+
+    left = 2.0
+    right = 3.0
+
+    @field_operator
+    def ternary_field_op(
+        a: Field[[IDim], float], b: Field[[IDim], float], left: float, right: float
+    ) -> Field[[IDim], float]:
+        return a if left < right else b
+
+    ternary_field_op(a, b, left, right, out=out, offset_provider={})
+    e = np.asarray(a) + np.asarray(b) if left < right else np.asarray(b)
+    np.allclose(e, out)
+
+    @field_operator
+    def ternary_field_op_scalars(left: float, right: float) -> Field[[IDim], float]:
+        return broadcast(3.0, (IDim,)) if left < right else broadcast(4.0, (IDim,))
+
+    ternary_field_op_scalars(left, right, out=out, offset_provider={})
+    e = np.full(e.shape, 3.0) if left < right else e
+    np.allclose(e, out)
+
+
+def test_ternary_operator_tuple():
+
+    size = 10
+    a = np_as_located_field(IDim)(np.ones((size,)))
+    b = np_as_located_field(IDim)(2 * np.ones((size,)))
+    out_1 = np_as_located_field(IDim)(np.zeros((size,)))
+    out_2 = np_as_located_field(IDim)(np.zeros((size,)))
+
+    left = 2.0
+    right = 3.0
+
+    @field_operator
+    def ternary_field_op(
+        a: Field[[IDim], float], b: Field[[IDim], float], left: float, right: float
+    ) -> tuple[Field[[IDim], float], Field[[IDim], float]]:
+        return (a, b) if left < right else (b, a)
+
+    # TODO(tehrengruber): directly call field operator when the generated programs support `out` being a tuple
+    @program
+    def ternary_field(
+        a: Field[[IDim], float],
+        b: Field[[IDim], float],
+        left: float,
+        right: float,
+        out_1: Field[[IDim], float],
+        out_2: Field[[IDim], float],
+    ):
+        ternary_field_op(a, b, left, right, out=(out_1, out_2))
+
+    ternary_field(a, b, left, right, out_1, out_2, offset_provider={})
+
+    e, f = (np.asarray(a), np.asarray(b)) if 2 < 3 else (np.asarray(b), np.asarray(a))
```

Comment:
Why not `left < right`?
```suggestion
    e, f = (np.asarray(a), np.asarray(b)) if left < right else (np.asarray(b), np.asarray(a))
```